en/setup/citationkeypatterns.md

```diff
@@ -18,57 +18,70 @@ The default key pattern is **`[auth][year]`**, and this could produce keys like
-* **`[title]`**:  Capitalize all the significant words of the title, and concatenate them. For example, `An awesome paper on JabRef` becomes `AnAwesomePaperonJabref`.
+* **`[shorttitle]`**: The first 3 words of the title, ignoring any function words \(see below\). For example, `An awesome paper on JabRef` becomes `AwesomePaperJabref`
+* **`[veryshorttitle]`**: The first word of the title, ignoring any function words \(see below\). For example, `An awesome paper on JabRef` becomes `Awesome`
+* **`[camel]`**: Capitalize and concatenate all the words of the title. For example, `An awesome paper on JabRef` becomes `AnAwesomePaperOnJabref`
+* **`[title]`**:  Capitalize all the significant words of the title, and concatenate them. For example, `An awesome paper on JabRef` becomes `AnAwesomePaperonJabref`
 
 JabRef considers the following words to be [function words](https://en.wikipedia.org/wiki/Function_word): "a", "an", "the", "above", "about", "across", "against", "along", "among", "around", "at", "before", "behind", "below", "beneath", "beside", "between", "beyond", "by", "down", "during", "except", "for", "from", "in", "inside", "into", "like", "near", "of", "off", "on", "onto", "since", "to", "toward", "through", "under", "until", "up", "upon", "with", "within", "without", "and", "but", "for", "nor", "or", "so", "yet".
 
 #### Other key patterns
 
 * **`[firstpage]`**: The number of the first page of the publication \(Caution: this will return the lowest number found in the pages field, i.e. for `7,41,73--97` it will return `7`.\)
-* **`[pageprefix]`**: The non-digit prefix of pages \(like "L" for L7\) or "" if no non-digit prefix exists \(like "" for `7,41,73--97`\) .
-* **`[keywordN]`**: Keyword number N from the “keywords” field, assuming keywords are separated by commas or semicolons.
+* **`[pageprefix]`**: The non-digit prefix of pages \(like "L" for L7\) or "" if no non-digit prefix exists \(like "" for `7,41,73--97`\) 
+* **`[keywordN]`**: Keyword number N from the “keywords” field, assuming keywords are separated by commas or semicolons
 * **`[lastpage]`**: The number of the last page of the publication \(See the remark on `firstpage`\)
 * **`[shortyear]`**: The last 2 digits of the publication year
 
+#### BibTeX fields
+
+In addition to the special field markers, most BibTeX, BibLaTeX, and JabRef field names can be used directly. If you regularly use a field marker not on this list, you are encouraged to add it.
+
+* **`[author]`** `Ada Lovelace and Charles Babbage` becomes `AdaLovelaceandCharlesBabbage`
+* **`[date]`** `2020-09-25`
+* **`[day]`** `02` becomes `2`
+* **`[groups]`** The groups or sgroups in JabRef. Subgroup `AppleTrees` and group `Trees` becomes `AppleTreesTrees`
+* **`[month]`** `03` becomes `March` 
+* **`[year]`** `2020`
+
+**Note:** More fields can be found in the source-code, but their use can lead to unexpected results and is, in general, discouraged. E.g., **`[abstract]`** can produce inconveniently long keys.
+
```

Comment:
Is it about the source-code of JabRef or about the library code?

Reply:
I would suggest saying that you can use any field of the entry type.
If I had defined a field acceded, I could use that as well

Reply:
And to both, sorry about the delayed response.
I am referring to the source-code, and I should have added a link to it, sorry 😛 

@Siedlerchr are you referring to https://docs.jabref.org/setup/customentrytypes ? I did not know about that feature, but yes, that will be included 😀

Reply:
@Siedlerchr do you know of a "typical" usage of this feature?

Reply:
A special field I would add to the list is `entrytype` which returns the entry type .e.g article or book

In the custom entry entry type you can add completely new types or add extra custom fields to existing entry types. 
E.g. I can add a field "MyCustomField" to article.

For the key generator it doesn't matter. There isn't a list of fixed fields, it can parse/resolve all fields for the current entry. 
So you can even manually add a field in the entry editor source tab  for that entry and use that for the pattern generations.
See in BracketedPattern the last else call to:
` return entry.getResolvedFieldOrAlias(FieldFactory.parseField(pattern), database).orElse(""); `

Therefore I would simply say: You can use all fields for the entry(type).

Reply:
Thank you for the expanded explanation!

I'll investigate where to put `entrytype` as it is both a special field marker and an `InternalField`. Perhaps it can be removed from the "special fields" and resolved by that last BracketedPattern call (and included in the appropriate list).

I'll add a note on that all fields can be used. I just want to make it clear that you can add new fields but that there is a priority in which field gets used (e.g., you can't use the same name as a special field marker because they will always get resolved first).

I think it would also be appropriate to change the heading from "BibTeX fields" -> "Bibentry fields"

Reply:
I would not really mention Internal Fields, while you can use them, they are solely for JabRef's internal use (except the entrytype), Theoretically you pobably could produce a stackoverflow if you use the internal field for the bibtexkey  in the keygenerator...

Reply:
Hum... Perhaps the `InternalField` should be removed from the usable fields, and the `entrytype` left as a special field in that case?

Reply:
I mean, I don't think leaving them as is is going to be an issue. It's just that I don't like leaving potential undocumented features.

Reply:
One should also keep in mind that the documentation is meant for "normal" users, so I wouldn't refer to the source code. I think the example you give are sufficient and illustrate the idea.

Reply:
Yes, leave entrytype as a special case. And simply write: `Note: You can use any (standard) field present in the entry. However, multi line fields like comment or abstract can produce unexpected results. Their use is discouraged`

Reply:
It is updated and a sentence on https://docs.jabref.org/setup/customentrytypes is added